streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -429,6 +421,15 @@ public void closeDirty() {
         log.info("Closed dirty");
     }
 
+    @Override
+    public void updateInputPartitions(final Set<TopicPartition> topicPartitions) {
+        super.updateInputPartitions(topicPartitions);
+        partitionGroup.updatePartitions(topicPartitions, recordQueueCreator::createQueue);
+        if (state() != State.RESTORING) { // if task is RESTORING then topology will be initialized in completeRestoration
```

Comment:
If the state was RUNNING then we would effectively call `initializeTopology` twice. Is that intentional?

Reply:
if I correctly remember if state==`RUNNING` `initializeTopology` is called only once in `update` method. Only if state == RESTORING it will be called later. It's no problem to call this method twice but it's aka optimization.